harness/tests/determined/common/experimental/test_workspace.py

```diff
@@ -71,3 +87,21 @@ def test_workspace_constructor_populates_id_from_name(
 def test_workspace_constructor_doesnt_populate_name_from_id(standard_session: api.Session) -> None:
     ws = workspace.Workspace(session=standard_session, workspace_id=1)
     assert ws.name is None
+
+
+@responses.activate
```

Comment:
I think the behavior under test here looks like `test_resource_pools_gets_single_resource_pool` or the like. You've set up the fixtures and managed `responses` well, but I'm not sure there's enough logic in the method itself for this specific behavior that it wants a unit test.

I'll need to come back to this after my next meeting, here. I think the key will be to think about the code path under exercise with this test for the code that's specifically in `Workspace.respource_pools`.

Reply:
The `resource_pools` function is so simple -- as syntactic sugar on `bindings`, it mostly just passes arguments through. The only part of it that has any trickiness for me is the pagination part. That part is hard to think about.

So I think to the extent you want to include any unit test for this method, testing that it actually does compile results across multiple pages is the only code I think is worth a unit test.

If this was my PR, I'd like to think that I'd include a test for it. But when the rubber hits the road I might end up being too lazy for it. Let's call it "recommended" but not "required."

If you want to, I've already done something very similar in [test_model.py](https://github.com/determined-ai/determined/blob/497df171c14ea3089925ba2880e273a9a5274f04/harness/tests/determined/common/experimental/test_model.py#L30) (added after discovering a bug in `get_versions`'s pagination).